sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgInstructionFactory.cs

```diff
@@ -491,10 +501,62 @@ private IEnumerable<Instruction> ProcessMemberAccessExpression(MemberAccessExpre
                     MethodId = identifier.ToString()
                 }
             };
-            instruction.Assigncall.Args.AddRange(arguments.Select(a => a.Expression));
+
+            IList<Instruction> newInstructions = new List<Instruction>();
+            var processedArgs = ProcessInstructionArguments(arguments, newInstructions);
```

Comment:
Not sure if we always pass a new List to the method. If so, maybe having a out param would be better.

Reply:
I've added the "if identifier is assignment check" back in so this has changed slightly.